core/ldap_api.php

```diff
@@ -180,6 +205,8 @@ function ldap_realname_from_username( $p_username ) {
  *
  * @param string $p_string The string to escape.
  * @return string The escaped string.
+ *
+ * @TODO: could we use built-in ldap_escape() instead of this custom code?
```

Comment:
This code exists (since 2009) because ldap_escape() was only introduced in PHP 5.6. Since we still support 5.5.9, I would suggest to update the function with a PHP version check, and use the existing code as fallback if PHP < 5.6, and built-in ldap_escape() if not.

Reply:
OK, that sounds easy enough.

Now that I look a bit more, I also think the existing code is not escaping everything it should be... will audit that...

Reply:
> the existing code is not escaping everything it should be... will audit that

Don't waste too much time on this, I'm planning to increase the minimum PHP requirement to 7.x, in 2.25.0 if I can, or in the next release if not ([#25956](https://mantisbt.org/bugs/view.php?id=25956)).

Reply:
OK, I've just removed this TODO for now...